packages/query-core/src/hydration.ts

```diff
@@ -21,6 +21,7 @@ export interface DehydrateOptions {
 
 export interface HydrateOptions {
   defaultOptions?: {
+    transformPromise?: (promise: Promise<any>) => Promise<any>
```

Comment:
can we set this to unknown?

```suggestion
    transformPromise?: (promise: Promise<unknown>) => Promise<unknown>
```

Reply:
could, although it messes with some serializers expected types so the user would have to `as any` it

![CleanShot 2024-06-11 at 11 08 10@2x](https://github.com/TanStack/query/assets/51714798/f0ea8c12-17f3-44ec-b389-fe31797dcf9d)

Reply:
hm okay

Reply:
could be accepted to do this - atleast then you're explicit that you're not certain that this data is serialized

![CleanShot 2024-06-11 at 11 29 27@2x](https://github.com/TanStack/query/assets/51714798/b8ce671d-fad3-475d-8066-75044e7648a2)